drivers/i2s/i2s_mcux_sai.c

```diff
@@ -953,58 +953,13 @@ static void audio_clock_settings(const struct device *dev)
 	uint32_t clock_name = (uint32_t) dev_cfg->clk_sub_sys;
 
 	/*Clock setting for SAI*/
-#if CONFIG_CLOCK_CONTROL_MCUX_CCM_REV2
-	switch (clock_name) {
-	case IMX_CCM_SAI1_CLK:
-		CLOCK_SetRootClockMux(kCLOCK_Root_Sai1, dev_cfg->clk_src);
-		CLOCK_SetRootClockDiv(kCLOCK_Root_Sai1, dev_cfg->clk_src_div);
-		break;
-	case IMX_CCM_SAI2_CLK:
-		CLOCK_SetRootClockMux(kCLOCK_Root_Sai2, dev_cfg->clk_src);
-		CLOCK_SetRootClockDiv(kCLOCK_Root_Sai2, dev_cfg->clk_src_div);
-		break;
-	case IMX_CCM_SAI3_CLK:
-		CLOCK_SetRootClockMux(kCLOCK_Root_Sai3, dev_cfg->clk_src);
-		CLOCK_SetRootClockDiv(kCLOCK_Root_Sai3, dev_cfg->clk_src_div);
-		break;
-	case IMX_CCM_SAI4_CLK:
-		CLOCK_SetRootClockMux(kCLOCK_Root_Sai4, dev_cfg->clk_src);
-		CLOCK_SetRootClockDiv(kCLOCK_Root_Sai4, dev_cfg->clk_src_div);
-		break;
-	default:
-		LOG_ERR("wrong clock system configured");
-		return;
-	}
-#endif
-
-#if CONFIG_CLOCK_CONTROL_MCUX_CCM
-	switch (clock_name) {
-	case IMX_CCM_SAI1_CLK:
-		CLOCK_SetMux(kCLOCK_Sai1Mux, dev_cfg->clk_src);
-		CLOCK_SetDiv(kCLOCK_Sai1PreDiv, dev_cfg->clk_pre_div);
-		CLOCK_SetDiv(kCLOCK_Sai1Div, dev_cfg->clk_src_div);
-		break;
-	case IMX_CCM_SAI2_CLK:
-		CLOCK_SetMux(kCLOCK_Sai2Mux, dev_cfg->clk_src);
-		CLOCK_SetDiv(kCLOCK_Sai2PreDiv, dev_cfg->clk_pre_div);
-		CLOCK_SetDiv(kCLOCK_Sai2Div, dev_cfg->clk_src_div);
-		break;
-	case IMX_CCM_SAI3_CLK:
-		CLOCK_SetMux(kCLOCK_Sai2Mux, dev_cfg->clk_src);
-		CLOCK_SetDiv(kCLOCK_Sai2PreDiv, dev_cfg->clk_pre_div);
-		CLOCK_SetDiv(kCLOCK_Sai2Div, dev_cfg->clk_src_div);
-		break;
-	default:
-		LOG_ERR("wrong clock system configured");
-		return;
-	}
-#endif
+	imxrt_audio_codec_pll_init(clock_name, dev_cfg->clk_src,
+				dev_cfg->clk_pre_div, dev_cfg->clk_src_div);
```

Comment:
Thank you for changing this. Instead of calling a function, can this clock setup be done inside clock_init function as was done for other peripherals with enclosing `CONFIG `wrappers.

Reply:
@mmahadevan108 , in clock_init we can not call the DTS settings of I2S which is configurable. so we just remove all the clock setting from dts and use fixed numbers? if so can we make it as another PR? this change is somehow big whcih will impact the whole i2s driver.

Reply:
@hakehuang I understand.